src/firstboot/firstboot.c

```diff
@@ -774,93 +777,68 @@ static int write_root_shadow(const char *shadow_path, const char *hashed_passwor
 static int process_root_args(void) {
         _cleanup_close_ int lock = -1;
         struct crypt_data cd = {};
-        const char *hashed_password;
-        const char *etc_shadow;
+        const char *password, *hashed_password;
+        const char *etc_passwd, *etc_shadow;
         int r;
 
+        etc_passwd = prefix_roota(arg_root, "/etc/passwd");
         etc_shadow = prefix_roota(arg_root, "/etc/shadow");
-        if (laccess(etc_shadow, F_OK) >= 0 && !arg_force)
+
+        /* Reduce the number of possible scenarios by always making sure we have permission to modify both
+         * /etc/passwd and /etc/shadow. */
+        if ((laccess(etc_passwd, F_OK) >= 0 || laccess(etc_shadow, F_OK) >= 0) && !arg_force)
```

Comment:
this is not a permssion check. It's an existance check, i.e. F_OK

Reply:
My reasoning was that we only have permission from the user to modify these files if they don't exist yet or arg_force is specified, I'll update the comment to something more clarifying